src/notecard_platform.c

```diff
@@ -55,6 +121,16 @@ esp_err_t notecard_platform_i2c_init(const notecard_i2c_config_t *config)
     esp_err_t ret = i2c_new_master_bus(&bus_config, &g_i2c_bus_handle);
     if (ret != ESP_OK) {
         ESP_LOGE(TAG, "I2C master bus creation failed: %s", esp_err_to_name(ret));
+#ifdef CONFIG_NOTECARD_I2C_MUTEX
+        if (g_i2c_mutex != NULL) {
+            vSemaphoreDelete(g_i2c_mutex);
+            g_i2c_mutex = NULL;
+        }
```

Comment:
Cannot be `NULL`. `init` would have already returned with an error.

```suggestion
        vSemaphoreDelete(g_i2c_mutex);
        g_i2c_mutex = NULL;
```